### PR TITLE
ISPN-14978 EXPIRE / ISPN-14979 EXPIREAT / ISPN-14980 EXPIRETIME / ISPN-14981 PEXPIRETIME

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14978 EXPIRE
https://issues.redhat.com/browse/ISPN-14979 EXPIREAT
https://issues.redhat.com/browse/ISPN-14980 EXPIRETIME
https://issues.redhat.com/browse/ISPN-14981 PEXPIRETIME

Also use TimeService instead of relying on system time for more reliable testing

### CODE REVIEW

**Comment by tristantarrant** [comment]
Failures are unrelated

**Comment by tristantarrant** [comment]
Failures are unrelated

**Review by jabolina — COMMENTED** [review]
Just some small stuff, and adding the entries in the .adoc.

File: server/resp/src/main/java/org/infinispan/server/resp/commands/generic/EXPIRE.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      super(-3, 1, -1, 1);
+      this.unixTime = at;
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+      byte[] key = arguments.get(0);
+      int expiration = ArgumentUtils.toInt(arguments.get(1));
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/generic/EXPIRE.java]: Fixed

File: server/resp/src/main/java/org/infinispan/server/resp/commands/generic/EXPIRE.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+   }
+
+   private final boolean unixTime;
+
+   public EXPIRE() {
+      this(false);
+   }
+
+   protected EXPIRE(boolean at) {
+      super(-3, 1, -1, 1);
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/generic/EXPIRE.java]: Fixed

**Comment by tristantarrant** [comment]
Fixed and rebased

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Integrated, thanks @tristantarrant!

**Pull request merged**